### ISSUE TITLE
How to debug/print the gradient calculated by torchrec or dmp?

### ISSUE THREAD

**Issue by ptx9363** [meta-pytorch/torchrec#1293]
As i known, the gradient is calculated and updated to weights in loss.backward(). How could i print/get the calculated gradient in the model for debugging my model.

**Comment by YLGH** [comment]
If you're using a fused optimizer unfortunantely you won't be able to get the gradients out for the embedding tables. However, you should be able to see them for everything else before you call optimizer.zero_grad.

**Comment by colin2328** [comment]
@ptx9363 , to add to what @YLGH  said
if inspecting grads is important to you, don't use the fused optimizer. This can be configured through ParameterConstraints.

In general, we recommend fused optimizer as it will greatly increase training perf.

Going to close the task, feel free to reopen if you have additional qs
